src/app/zap-templates/templates/chip/CHIPClusters-src.zapt

```diff
@@ -35,7 +38,11 @@ CHIP_ERROR {{asCamelCased clusterName false}}Cluster::{{asCamelCased name false}
     uint8_t argSeqNumber = 0;
 {{/first}}
     // {{asCamelCased label}}: {{asCamelCased type}}
-{{#if (isString type)}}
+{{#if (isByteString type)}}
+    // Check the size of the buffer, TLV::Write currently takes a uint32_t instead of size_t.
+    VerifyOrReturnError({{asCamelCased label}}.Length() <= UINT32_MAX, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Depending on platform, this can trigger "this comparison is always true" warnings that will lead to compilation failures.  Please use `CanCastTo`; that's why we have it.

Reply:
Removed this line, and TLVWriter can take care of this now.